pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogicCCL_select_for_update_read_committed failed

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogicCCL_select_for_update_read_committed [failed](https://tanzanite.cluster.engflow.com/invocations/default/857d483b-f240-48cb-bd60-f7a1e208a283?testReportRun=23&testReportShard=42&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on master @ [a0a52309cbbe5cb68d576fb366411fa7d7cf3574](https://github.com/cockroachdb/cockroach/commits/a0a52309cbbe5cb68d576fb366411fa7d7cf3574):


```
=== RUN   TestTenantLogicCCL_select_for_update_read_committed
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantLogicCCL_select_for_update_read_committed2979869532
    test_log_scope.go:81: use -show-logs to present logs inline
    logic.go:4317: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogicCCL_select_for_update_read_committed2979869532
--- FAIL: TestTenantLogicCCL_select_for_update_read_committed (2.72s)
=== RUN   TestTenantLogicCCL_select_for_update_read_committed/skip_locked
    logic.go:2967: 
         
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/select_for_update_read_committed:610: SELECT * FROM xyz WHERE z = 100 ORDER BY x FOR UPDATE SKIP LOCKED
        expected:
            2  10  100
            
        but found (query options: "") :
            
    logic.go:2205: 
         /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/select_for_update_read_committed:615: too many errors encountered, skipping the rest of the input
[12:33:34] --- done: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/select_for_update_read_committed with config 3node-tenant: 107 tests, 2 failures
[12:33:34] --- total progress: 107 statements
--- total: 107 tests, 2 failures
    --- FAIL: TestTenantLogicCCL_select_for_update_read_committed/skip_locked (0.04s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=23</code>
 - <code>shard=42</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogicCCL_select_for_update_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-40912

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogicCCL_select_for_update_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16334168?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16334168?buildTab=artifacts#/) on master @ [dbb0dc367e28cde4064d94e0dda6c06a4060dddf](https://github.com/cockroachdb/cockroach/commits/dbb0dc367e28cde4064d94e0dda6c06a4060dddf):


```
=== RUN   TestTenantLogicCCL_select_for_update_read_committed
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/8d47336da04684d74b581ec995900e36/logTestTenantLogicCCL_select_for_update_read_committed1195371045
    test_log_scope.go:81: use -show-logs to present logs inline
    logic.go:4317: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/8d47336da04684d74b581ec995900e36/logTestTenantLogicCCL_select_for_update_read_committed1195371045
--- FAIL: TestTenantLogicCCL_select_for_update_read_committed (4.34s)
=== RUN   TestTenantLogicCCL_select_for_update_read_committed/skip_locked
    logic.go:2967: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15508/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/select_for_update_read_committed:610: SELECT * FROM xyz WHERE z = 100 ORDER BY x FOR UPDATE SKIP LOCKED
        expected:
            2  10  100
            
        but found (query options: "") :
            
    logic.go:2205: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15508/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/select_for_update_read_committed:615: too many errors encountered, skipping the rest of the input
[16:17:03] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15508/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/select_for_update_read_committed with config 3node-tenant: 107 tests, 2 failures
[16:17:04] --- total progress: 107 statements
--- total: 107 tests, 2 failures
    --- FAIL: TestTenantLogicCCL_select_for_update_read_committed/skip_locked (0.09s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogicCCL_select_for_update_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/ccl/logictestccl/testdata/logic_test/select_for_update_read_committed
+++ b/pkg/ccl/logictestccl/testdata/logic_test/select_for_update_read_committed
@@ -579,6 +579,9 @@ statement ok
 SET optimizer_use_lock_op_for_serializable = true
 
 statement ok
+SET enable_durable_locking_for_serializable = true
+
+statement ok
 BEGIN ISOLATION LEVEL SERIALIZABLE
 
 # Lock the first row.
@@ -606,6 +609,9 @@ COMMIT;
 statement ok
 SET optimizer_use_lock_op_for_serializable = true
 
+statement ok
+SET enable_durable_locking_for_serializable = true
+
 query III
 SELECT * FROM xyz WHERE z = 100 ORDER BY x FOR UPDATE SKIP LOCKED
 ----
@@ -619,7 +625,13 @@ COMMIT
 statement ok
 RESET optimizer_use_lock_op_for_serializable
 
+statement ok
+RESET enable_durable_locking_for_serializable
+
 user root
 
 statement ok
 RESET optimizer_use_lock_op_for_serializable
+
+statement ok
+RESET enable_durable_locking_for_serializable

